frappe/desk/form/load.py

```diff
@@ -43,7 +40,8 @@ def getdoc(doctype, name, user=None):
 		# add file list
 		doc.add_viewed()
 		get_docinfo(doc)
-
+	except frappe.DoesNotExistError:
+		return []
```

Comment:
Cannot rely on this, `frappe.throw` shows an error modal which is why UI tests are failing. Why is this required? What was the issue with the previous handling?
<img width="1202" alt="Screenshot 2021-03-18 at 9 05 48 AM" src="https://user-images.githubusercontent.com/13928957/111569683-31495d80-87c9-11eb-9509-8bbdc5c0d194.png">

Reply:
frappe.db.exists makes a db call, in case of virtual DocType we wont have a table so it will always return empty.

Reply:
I was not able to reproduce the error which you mentioned
![contack](https://user-images.githubusercontent.com/11792643/111579996-edf8ea00-87dc-11eb-8b14-463121718376.gif)

Reply:
what do you think is the best way to handle this issue?

Reply:
Can we have a `exists` method for virtual doctype as well? `frappe.db.exists` can internally use that method to check if virtual document exists.

Reply:
I don't think its a correct way to handle. Example consider if virtual doctype's  backend is an api then we might end up calling same api twice ones to check if exists and second time to actually get the data.

Reply:
Can we set frappe.local.message_log to false here?

Reply:
> first call to check if document exists and second call to get the data.

I think that's fine... but it will help in other cases where we are just using `frappe.db.exists` for different validations.

Reply:
`frappe.db.exists` internally uses `frappe.db.get_value`  so I have added support for `frappe.db.get_value` in virtual doctype,